Source/Atomic/UI/UIFinderWindow.cpp

```diff
@@ -0,0 +1,618 @@
+//
+// Copyright (c) 2014-2017, THUNDERBEAST GAMES LLC All rights reserved
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include <TurboBadger/tb_widgets.h>
+#include <TurboBadger/tb_widgets_common.h>
+
+#include <Atomic/IO/FileSystem.h>
+#include <Atomic/IO/Log.h>
+#include <Atomic/IO/File.h>
+
+#include "UI.h"
+#include "UIEvents.h"
+#include "UIWindow.h"
+#include "UIEditField.h"
+#include "UISelectList.h"
+#include "UIPromptWindow.h"
+#include "UIFinderWindow.h"
+#include "UISelectItem.h"
+#include "UIMenuWindow.h"
+
+using namespace tb;
+
+namespace Atomic
+{
+
+/// finder window
+UIFinderWindow::UIFinderWindow(Context* context, UIWidget* target, const String& id, bool createWidget) : 
+    UIWindow(context, false),
+    finderMode(0),
+    currentPath(),
```

Comment:
It is an effort to reduce unintended behavior. In a former life, this initialization was required to prevent bugs from uninitialized values, Ive heard on windows, variables are automatically set to 0, but this is not the case on linux and probably osx, android.